plugins/vpp/wireguardplugin/vppcalls/vpp2009/dump_vppcalls.go

```diff
@@ -0,0 +1,94 @@
+//  Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package vpp2009
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net"
+
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/binapi/vpp2009/ip_types"
+	vpp_wg "go.ligato.io/vpp-agent/v3/plugins/vpp/binapi/vpp2009/wireguard"
+	wg "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/wireguard"
+)
+
+// DumpWgPeers implements wg handler.
+func (h *WgVppHandler) DumpWgPeers() (peerList []*wg.Peer, err error) {
+	req := &vpp_wg.WireguardPeersDump{}
+	requestCtx := h.callsChannel.SendMultiRequest(req)
+
+	var vppPeerList []*vpp_wg.WireguardPeersDetails
+	for {
+		vppPeerDetails := &vpp_wg.WireguardPeersDetails{}
+		stop, err := requestCtx.ReceiveReply(vppPeerDetails)
+		if stop {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+		vppPeerList = append(vppPeerList, vppPeerDetails)
+	}
+
+	for _, vppPeerDetails := range vppPeerList {
+		peerDetails := &wg.Peer{
+			Port:                uint32(vppPeerDetails.Peer.Port),
+			PersistentKeepalive: uint32(vppPeerDetails.Peer.PersistentKeepalive),
+			Flags:               uint32(vppPeerDetails.Peer.Flags),
+		}
+
+		peerDetails.PublicKey = base64.StdEncoding.EncodeToString(vppPeerDetails.Peer.PublicKey)
+
+		allowedIpsDetails := vppPeerDetails.Peer.AllowedIps
+		var i uint8
+		for i = 0; i < vppPeerDetails.Peer.NAllowedIps; i++ {
+			prefix := allowedIpsDetails[i]
+			var ipAddr string
+			ipByte := make([]byte, 16)
+			copy(ipByte[:], prefix.Address.Un.XXX_UnionData[:])
+			if prefix.Address.Af == ip_types.ADDRESS_IP6 {
+				ipAddr = fmt.Sprintf("%s/%d", net.IP(ipByte).To16().String(), uint32(prefix.Len))
+			} else {
+				ipAddr = fmt.Sprintf("%s/%d", net.IP(ipByte[:4]).To4().String(), uint32(prefix.Len))
+			}
+			peerDetails.AllowedIps = append(peerDetails.AllowedIps, ipAddr)
+		}
```

Comment:
`len(AllowedIps)` is same as `NAllowedIps` and `ip_types.Prefix` has `String()` method for converting to string so you can simplify this:

```suggestion
for _, prefix := range vppPeerDetails.Peer.AllowedIps {
    peerDetails.AllowedIps = append(peerDetails.AllowedIps, prefix.String()
}
```